jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -100,7 +101,10 @@ private static String capitalizeFirstLetter(String string) {
     this(
         ImageConfiguration.builder(baseImage.getImageReference())
             .setDockerClient(
-                new DockerClient(baseImage.getDockerExecutable(), baseImage.getDockerEnvironment()))
+                DockerClientResolver.resolve(baseImage.getDockerEnvironment())
+                    .orElse(
+                        new CliDockerClient(
+                            baseImage.getDockerExecutable(), baseImage.getDockerEnvironment())))
```

Comment:
This could use an option to pass the client explicitly

Reply:
@eddumelendez ping :)

Reply:
`Jib.from(DockerClient, DockerDaemonImage)` will use `JibContainerBuilder(DockerClient, DockerDaemonImage)`

https://github.com/GoogleContainerTools/jib/pull/3703/files#diff-7660df27e5571633351e3a238f5308665ea1ff9977148d63fba26ba44e0a0e90R123-R129

  https://github.com/GoogleContainerTools/jib/pull/3703/files#diff-ec17eab0abae7c154524f4395bb20d8979b1aa46b44806ceb141e27790e83191R135-R144